src/components/HistoryTables/MessageHistory/hooks/useMethodName.ts

```diff
@@ -1,46 +1,25 @@
-import { useMemo } from 'react'
-import { useActorQuery } from '../../../../generated/graphql'
-import { getMethodName } from '../../methodName'
-import { decodeActorCID } from '../../../../utils'
-import { MessageConfirmedRow, MessagePendingRow } from '../../types'
-import convertAddrToPrefix from '../../../../utils/convertAddrToPrefix'
-import {
-  useEnvironment,
-  useLogger
-} from '../../../../services/EnvironmentProvider'
+import { getActorName, getMethodName } from '@glif/filecoin-actor-utils'
 
-type MessageForMethodNameType =
-  | Pick<MessageConfirmedRow, 'to' | 'cid' | 'method'>
-  | (MessagePendingRow & { actorName: string })
+import { Address, useActorQuery } from '../../../../generated/graphql'
+import convertAddrToPrefix from '../../../../utils/convertAddrToPrefix'
+import { useEnvironment } from '../../../../services/EnvironmentProvider'
 
 export const useMethodName = (
-  message: MessageForMethodNameType
-): { methodName: string; actorName: string } => {
+  address: Address,
+  method: string
+): string | null => {
   const { coinType, networkName } = useEnvironment()
-  const logger = useLogger()
-  const actor = useActorQuery({
+
+  // Get actor data from GraphQL
+  const { data } = useActorQuery({
     variables: {
-      address: convertAddrToPrefix(
-        message?.to.robust || message?.to.id,
-        coinType
-      )
-    }
+      address: convertAddrToPrefix(address?.robust || address?.id, coinType)
+    },
+    skip: !address
```

Comment:
Technically, if no `id` address exists, the `useActorQuery` will return null. I think this is fine for now, but we _could_ be more specific about when to skip in the future. An `id` gets assigned when a `robust` (stable) address has some interaction with the chain

Reply:
Additional future optimization - we know that an `f1` or an `f3` actor _must_ be an `account` actor. An `f2` address is a non-account actor